File: apidemo/sensordemo/environmentalsensordemo/src/main/java/com/gdkdemo/sensor/environmental/cp/core/IDPair.java
package com.gdkdemo.sensor.environmental.cp.core;


public final class IDPair
{
    private final Long id;
    private final String guid;

    public IDPair(Long id, String guid)
    {
        // TBD: Validation?
        this.id = id;
        this.guid = guid;
    }
    
    public Long id()
    {
        return id;
    }

    public String guid()
    {
        return guid;
    }
    
    public int hashCode()
    { 
        int hashId = id == null ? 0 : id.hashCode(); 
        int hashGuid = guid == null ? 0 : guid.hashCode(); 
        return (hashId + hashGuid) * hashGuid + hashId; 
    } 
 
    @Override
    public String toString() 
    {  
        return "(" + id + ", " + guid + ")";  
    }

}

File: xamarindemo/sensordemo/EnvironmentalSensorDemo/Cp/Core/IDPair.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace EnvironmentalSensorDemo.Cp.Core
{
    public class IDPair
    {
        private long _id;
        private String _guid;

        public IDPair(long id, String guid)
        {
            // TBD: Validation?
            this._id = id;
            this._guid = guid;
        }

        public long id()
        {
            return _id;
        }

        public string guid()
        {
            return _guid;
        }

        public int hashCode()
        {
            int hashId = _id == null ? 0 : _id.GetHashCode();
            int hashGuid = _guid == null ? 0 : _guid.GetHashCode();
            return (hashId + hashGuid)*hashGuid + hashId;
        }

        public override string ToString()
        {
            return "(" + _id + ", " + _guid + ")";
        }

    }
}